src/py21cmfast/src/Find_Haloes.c

```diff
@@ -0,0 +1,470 @@
+
+// Re-write of find_halos.c from the original 21cmFAST
```

Comment:
Justa nit-pick, but can the name of this file please be `find_haloes.c` (no capitals)?

Reply:
Do you mean re-naming my existing file "Find_Haloes.c" to "find_haloes.c". If this is true nit-picking, wouldn't that imply you would want all .c files to be renamed as lower case?

Reply:
The python style is to name modules in `snake_case`, and classes in `CamelCase`. This is C, and I don't know the typical style there. But a mixture of `Snake_Camel` is definitely odd to me. I think at the moment the file-naming convention we seem to have arrived at is that non-public C files (tools and such) are `snake_case` and API-level files seem to be `CamelCase`. I'm happy if we stick with that. Also happy to make all filenames `snake_case`.